plugin/pkg/auth/authorizer/rbac/rbac.go

```diff
@@ -62,7 +62,7 @@ type authorizingVisitor struct {
 func (v *authorizingVisitor) visit(source fmt.Stringer, rule *rbac.PolicyRule, err error) bool {
 	if rule != nil && RuleAllows(v.requestAttributes, rule) {
 		v.allowed = true
-		v.reason = fmt.Sprintf("allowed by %s", source.String())
+		v.reason = fmt.Sprintf("RBAC: allowed by %s", source.String())
```

Comment:
I'd like to be able to tell which authorizer allowed a request. This approach is not reliable, and would require updating every reason message.

Instead, I suggest:
1) Add a `Name()` function to the Authorizer interface
2) Update the union authorizer to annotate the audit log with the decider's name

Reply:
not sure about this approach

> Add a Name() function to the Authorizer interface

It's unclear what `Name()` would be for the union authorizer

> Update the union authorizer to annotate the audit log with the decider's name

I didn't think we wanted to plumb audit events down into all the authorizers

that also means you lose visibility to what authorizer made the decision if it was on the other side of a webhook boundary

Reply:
> It's unclear what Name() would be for the union authorizer
`union`? Or just name the authorizers when they're passed to the union authorizer.

> I didn't think we wanted to plumb audit events down into all the authorizers
good point. maybe include it in the reason then? Or is that potentially leaking too much information to the requester?

> that also means you lose visibility to what authorizer made the decision if it was on the other side of a webhook boundary
The name doesn't need to be a constant. The webhook could derive its name from the config (although that means you can't have multiplexed authorizers on the other side).